Source/WebCore/Modules/websockets/WebSocket.cpp

```diff
@@ -334,7 +334,7 @@ ExceptionOr<void> WebSocket::connect(const String& url, const Vector<String>& pr
 
     auto reportRegistrableDomain = [domain = RegistrableDomain(m_url).isolatedCopy()](auto& context) mutable {
         if (RefPtr frame = downcast<Document>(context).frame())
-            frame->loader().client().didLoadFromRegistrableDomain(WTFMove(domain));
+            frame->checkedLoader()->client().didLoadFromRegistrableDomain(WTFMove(domain));
```

Comment:
This is odd. `client()` is clearly a trivial function. We shouldn't need to use `checkedLoader` here.

Reply:
client() is trivial but didLoadFromRegistrableDomain is not. A similar change was done in 273326@main. Shouldn't transitivity apply here?

Reply:
No. If didLoadFromRegistrableDomain is non-trivial, then client() is the one that needs to return Ref/RefPtr variant.

Reply:
The client is stored using a unique_ptr in FrameLoader, so I feel it is basically part of FrameLoader and FrameLoader still needs to be checked on these operations. @cdumez  WDYT? I guess we should either adopt the new changes or undo the existing ones, I am fine with doing it either way, just let me know.

Reply:
A better approach is for FrameLoaderClient to have ref/deref functions that forwards ref/deref to FrameLoader.

Reply:
I see what you mean. I started implementing that, however it is cumbersome given the FrameLoaderClient inheritance structure and the number of call sites to update. What do you think about my latest revision that follows the BackForwardController solution?